app.py

```diff
@@ -31,46 +30,158 @@ def init_app(self):
         }
         
         self.engine.register_command("show_in_filesystem", self.show_in_filesystem, p)
-            
+
     def launch(self, path):
-        self.log_debug("Launching file system viewer for folder %s" % path)        
-        
-        # get the setting        
+        """
+        Given a path this method will choose the appropriate way to open that as a folder in the OS's default
+        file manager. If the path to the file is invalid then nothing will happen.
+        :param path: A path to a file or folder.
+        """
+
+        if os.path.isfile(path):
+            self._launch_filemanager_for_file(path)
+        elif os.path.isdir(path):
+            self._launch_filemanager_for_folder(path)
+        else:
+            # possibly we have an image sequence and therefore its a symbolic path, instead check to see if
+            # the parent folder of the path is valid and try opening that.
+            # todo: The issue with this logic is that possibly it was a directory that just didn't exist,
+            # so we would just be gathering the next directory up,
+            # ideally need a better way to handle sequences.
+            parent_dir = os.path.dirname(path)
+            self._launch_filemanager_for_folder(parent_dir)
+
+            
+    def _launch_filemanager_for_folder(self, path):
+        """
+        This method will take a path to a folder and open it in the OS's default file manager.
+        :param path: A folder path
+        :raises: Exception: if the Platform is not supported
+        :raises: ValueError: if the path is not a valid directory.
+        """
+        self.log_debug("Launching file system viewer for folder %s" % path)
+
+        # Check that we don't have a file path.
+        if not os.path.isdir(path):
+            raise ValueError(
+                "The path \"%s\" is not a valid directory." % path)
+
+        # get the setting
         system = sys.platform
-        
+
         # run the app
-        if system == "linux2":
+        if system.startswith("linux2"):
```

Comment:
Do a startswith("linux"). There are some Linuxes where linux3 is returned.

Yeah, we should have a method in core that detects that...